planner/core/optimizer.go

```diff
@@ -297,10 +298,29 @@ func DoOptimize(ctx context.Context, sctx sessionctx.Context, flag uint64, logic
 }
 
 // refineCETrace will adjust the content of CETrace.
-// Currently, it will (1) deduplicate trace records and (2) fill in the table name.
+// Currently, it will (1) deduplicate trace records, (2) sort the trace records (to make it easier in the tests) and (3) fill in the table name.
 func refineCETrace(sctx sessionctx.Context) {
 	stmtCtx := sctx.GetSessionVars().StmtCtx
 	stmtCtx.OptimizerCETrace = tracing.DedupCETrace(stmtCtx.OptimizerCETrace)
+	sort.Slice(stmtCtx.OptimizerCETrace, func(i, j int) bool {
```

Comment:
should we need to control this to be called only in the test?

Reply:
I think it's ok to always sort it.
This function itself will only be called in `trace plan` statement, which is almost only used in the tests.
If we want to restrict the sort only in the tests, it's also not easy to implement it. Because the explaintest itself is run from the outside of tidb, unlike the unit tests.

Reply:
got it